core/src/main/java/io/grpc/NameResolver.java

```diff
@@ -262,6 +320,29 @@ public SynchronizationContext getSynchronizationContext() {
       throw new UnsupportedOperationException("should have been implemented");
     }
 
+    /**
+     * Updates the {@link Channel} with the name resolution result.  This should be called from
+     * the name resolver once the servers have been resolved.  Additionally, this also includes
+     * the Service Config associated with the servers, if supported.
+     *
+     * @param result The full result of name resolution.
+     * @since 1.20.0
+     */
+    public void updateNameResolutionResult(Result result) {
+      throw new UnsupportedOperationException("Not implemented");
+    }
+
+    /**
+     * Updates the {@link Channel} with an error from the resolver. The caller is responsible for
+     * eventually invoking {@link NameResolver#refresh()} to re-attempt resolution.
+     *
+     * @param error a non-OK status
+     * @since 1.20.0
+     */
+    public void updateNameResolutionError(Status error) {
+      throw new UnsupportedOperationException("Not implemented");
+    }
```

Comment:
Since we have the `Observer`, these two methods should not be added.

Reply:
Sorry, removed.